woodwork/logical_types.py

```diff
@@ -83,7 +83,7 @@ def validate(self, series, *args, **kwargs):
         valid_dtype = self._get_valid_dtype(type(series))
         if valid_dtype != str(series.dtype):
             raise TypeValidationError(
-                f"Series dtype '{series.dtype}' is incompatible with {self.type_string} dtype.",
+                f"Series dtype '{series.dtype}' is incompatible with {self.type_string} LogicalType.",
```

Comment:
Should we include `should be {valid_dtype}` at the end of the message?

Reply:
Yep, we can

Reply:
Changed it to  `try converting to {valid_dtype}` to try to be consistent w/ the error msg in `column_accessor.py`. Not sure which is better